scapy/layers/inet.py

```diff
@@ -1003,8 +1003,8 @@ def _packetlist_timeskew_graph(self, ip, **kargs):
     """Tries to graph the timeskew between the timestamps and real time for a given ip"""
 
     # Filter TCP segments which source address is 'ip'
-    res = [self._elt2pkt(x) for x in self.res]
-    b = filter(lambda x:x.haslayer(IP) and x.getlayer(IP).src == ip and x.haslayer(TCP), res)
+    res = (self._elt2pkt(x) for x in self.res)
```

Comment:
Done. I kept the two lines, renamed res to tmp and moved to the new layer syntax.